packages/@lwc/template-compiler/src/parser/index.ts

```diff
@@ -771,6 +771,10 @@ export default function parse(source: string, state: State): TemplateParseResult
                 removeAttribute(element, name);
             }
         });
+
+        if ((!state.shouldScopeFragmentId && element.props?.id) || element.attrs?.id) {
```

Comment:
im having trouble to understand this condition... did you meant `!state.shouldScopeFragmentId && (element.props?.id || element.attrs?.id)`?

Reply:
Good catch, prettier tricked me with this. The `&&` operator has higher precedence than the `||` operator.